src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs

```diff
@@ -472,6 +532,105 @@ TypeSymbol getResultType(ExpressionSyntax node, TypeSymbol leftType, BindingDiag
             {
                 return ResultIsUsed(node) ? leftType : GetSpecialType(SpecialType.System_Void, diagnostics, node);
             }
+
+            // This method returns result in two ways:
+            // - If it has a result due to instance extensions, it returns ready to use BoundCompoundAssignmentOperator 
+            // - If it has static extensions result, it returns information via out parameters (staticBest, staticResultKind, staticOriginalUserDefinedOperators). 
+            BoundCompoundAssignmentOperator? tryApplyUserDefinedExtensionOperator(
+                AssignmentExpressionSyntax node,
+                BinaryOperatorKind kind,
+                bool tryInstance,
+                bool checkOverflowAtRuntime,
+                string staticOperatorName1,
+                string? staticOperatorName2Opt,
+                string? checkedInstanceOperatorName,
+                string? ordinaryInstanceOperatorName,
+                BoundExpression left,
+                BoundExpression right,
+                BindingDiagnosticBag diagnostics,
+                out BinaryOperatorAnalysisResult? staticBest,
+                out LookupResultKind staticResultKind,
+                out ImmutableArray<MethodSymbol> staticOriginalUserDefinedOperators)
+            {
+                staticBest = null;
+                staticResultKind = LookupResultKind.Empty;
+                staticOriginalUserDefinedOperators = [];
+
+                if (!this.Compilation.LanguageVersion.AllowNewExtensions())
+                {
+                    return null;
+                }
+
+                var result = BinaryOperatorOverloadResolutionResult.GetInstance();
+                CompoundUseSiteInfo<AssemblySymbol> useSiteInfo = GetNewCompoundUseSiteInfo(diagnostics);
+                var extensionDeclarationsInSingleScope = ArrayBuilder<NamedTypeSymbol>.GetInstance();
+                BoundCompoundAssignmentOperator? inPlaceResult = null;
+                AnalyzedArguments? analyzedArguments = null;
+
+                foreach (var scope in new ExtensionScopes(this))
+                {
+                    extensionDeclarationsInSingleScope.Clear();
```

Comment:
I'd missed in previous PRs: just to confirm, we don't need to clear `result` too?

Reply:
>I'd missed in previous PRs: just to confirm, we don't need to clear ```result``` too?

```BinaryOperatorExtensionOverloadResolutionInSingleScope``` clears it before repopulating.